### PR TITLE
Deprecated the telemetry configuration item and remove the telemetry code usage.

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#19719

Changelog：
- Mark the telemetry configuration item of each module as deprecated.
- Mark the module `azure-spring-cloud-telemetry` as deprecated, and remove the source code.
- Remove telemetry code of spring cloud module.

Some more context here:

Azure Spring libraries used to collect telemetry data by sending them to a hosted telemetry track endpoint, but now with using track 2 Azure SDks, we could set some spring flags in the HTTP user-agent header. So now there's no need of such telemetry libraries except for libraries not developed by Azure SDK, such as the redis integration.

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/cosmos/CosmosProperties.java [hunk 1]
```diff
@@ -102,6 +103,9 @@ public void setConsistencyLevel(ConsistencyLevel consistencyLevel) {
         this.consistencyLevel = consistencyLevel;
     }
 
+    @Deprecated
+    @DeprecatedConfigurationProperty(
+        reason = "Remove data collection based on fixed endpoints, and support based on service endpoints.")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/cosmos/CosmosProperties.java:108]: Same here.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/keyvault/KeyVaultProperties.java [hunk 2]
```diff
@@ -107,6 +108,9 @@ public void setCaseSensitiveKeys(String caseSensitiveKeys) {
         this.caseSensitiveKeys = caseSensitiveKeys;
     }
 
+    @Deprecated
+    @DeprecatedConfigurationProperty(
+        reason = "Remove data collection based on fixed endpoints, and support based on service endpoints.")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/keyvault/KeyVaultProperties.java:113]: Same here.

File: sdk/spring/azure-spring-cloud-autoconfigure/pom.xml [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
   <properties>
     <jacoco.min.linecoverage>0.10</jacoco.min.linecoverage>
-    <jacoco.min.branchcoverage>0.15</jacoco.min.branchcoverage>
+    <jacoco.min.branchcoverage>0.10</jacoco.min.branchcoverage>
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-autoconfigure/pom.xml:22]: Why are we decreasing this?

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java [hunk 4]
```diff
@@ -34,12 +32,6 @@
 @ConditionalOnClass(RedisOperations.class)
 @EnableConfigurationProperties(AzureRedisProperties.class)
 public class AzureRedisAutoConfiguration {
-    private static final String REDIS = "Redis";
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java:37]: There's no way to collect redis usage now.

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 5]
```diff
@@ -65,11 +63,4 @@ public TokenCredential credential(Environment environment) {
                                                    .alternativePrefix(AzureProperties.PREFIX)
                                                    .build();
     }
-
-    @Bean
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:69]: Why did you delete this?

File: sdk/spring/azure-spring-cloud-storage/pom.xml [hunk 6]
```diff
@@ -16,7 +16,7 @@
   <name>Azure Spring Cloud Storage</name>
 
   <properties>
-    <jacoco.min.linecoverage>0.3</jacoco.min.linecoverage>
+    <jacoco.min.linecoverage>0.1</jacoco.min.linecoverage>
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-storage/pom.xml:19]: Same here, should we add more test cases?

File: sdk/spring/azure-spring-cloud-telemetry/CHANGELOG.md [hunk 7]
```diff
@@ -1,35 +0,0 @@
-# Release History
-
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-telemetry/CHANGELOG.md:2]: How about keeping this doc and adding the depreciation here too?

File: sdk/spring/azure-spring-cloud-telemetry/README.md [hunk 8]
```diff
@@ -1,15 +1,9 @@
 # Azure Spring cloud azure telemetry client library for Java
-This package is an internal package sending telemetry data for Azure Spring Cloud services.
-
-## Getting started
-
-### Prerequisites
-- [Environment checklist][environment_checklist]
+We are deprecating this starter, Use service-based endpoints to collect data.
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-telemetry/README.md:2]: Not accurate too.

**Review by saragluna — CHANGES_REQUESTED** [review]
One more thing, do we delete the entry defined in the versioning client text file too?

File: sdk/spring/azure-spring-cloud-autoconfigure/pom.xml (hunk 3 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-autoconfigure/pom.xml:22]: Coverage checks have not been met.

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java (hunk 4 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java:37]: So, we will keep the telemetry usage of redis first?

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java (hunk 5 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:69]: Because of the he interface `SubscriptionSupplier` is under the `azure-spring-cloud-telemetry` module.

Comment (saragluna) [reply sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:69]: I see.

File: sdk/spring/azure-spring-cloud-storage/pom.xml (hunk 6 above)
Comment (saragluna) [reply sdk/spring/azure-spring-cloud-storage/pom.xml:19]: @moarychan let's try to add them instead of decrease the number here.

File: sdk/spring/azure-spring-cloud-telemetry/CHANGELOG.md [hunk 9]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
 ## 2.6.0-beta.1 (Unreleased)
-
+### Breaking Changes
+-- Deprecate the telemetry endpoint and use HTTP header User Agent instead.
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-telemetry/CHANGELOG.md:5]: Wrong format here, one `-` is enough.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-cloud-autoconfigure/pom.xml (hunk 3 above)
Comment (saragluna) [reply sdk/spring/azure-spring-cloud-autoconfigure/pom.xml]: Let's try to add more tests here.

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java (hunk 4 above)
Comment (saragluna) [reply sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cache/AzureRedisAutoConfiguration.java:37]: Let's remove the telemetry in this PR and figure out how to collect the Redis data later.

**Review by saragluna — APPROVED** [review]
LGTM.
